### PR TITLE
WebCore mac WIP

### PR DESCRIPTION
**Pull request by mnutt**
A few changes towards getting WebCore to build on MacOS. It doesn't build quite yet.

### CODE REVIEW

File: Source/WebCore/platform/network/DNSResolveQueue.cpp [hunk 1]
```diff
@@ -31,10 +31,10 @@
 #include "DNSResolveQueueSoup.h"
 #elif USE(CURL)
 #include "DNSResolveQueueCurl.h"
-#elif USE(CF)
```
Comment (mnutt) [inline Source/WebCore/platform/network/DNSResolveQueue.cpp]: Not sure if I should have done this, or to pull `DNSResolveQueueCFNet.h` into `Source/WebCore/PlatformQt.cmake`.

File: Source/WebCore/page/Crypto.cpp [hunk 2]
```diff
@@ -58,7 +58,7 @@ ExceptionOr<void> Crypto::getRandomValues(ArrayBufferView& array)
         return Exception { TypeMismatchError };
     if (array.byteLength() > 65536)
         return Exception { QuotaExceededError };
-#if OS(DARWIN)
+#if OS(DARWIN) && ENABLE(WEB_CRYPTO)
```
Comment (mnutt) [inline Source/WebCore/page/Crypto.cpp]: Should this just be `&& !PLATFORM(QT)` ?

Comment (annulen) [reply Source/WebCore/page/Crypto.cpp]: I've changed default settings and disabled WEB_CRYPTO for now.

File: Source/WebCore/platform/network/DNSResolveQueue.cpp (hunk 1 above)
Comment (annulen) [reply Source/WebCore/platform/network/DNSResolveQueue.cpp]: This is correct

**Comment by annulen** [comment]
I've merged your patches except Crypto.cpp as it can be delayed, and used BUILDING_QT__ in  JSC_API_AVAILABLE

**Comment by annulen** [comment]
Thanks!

**Comment by mnutt** [comment]
Thanks! Regarding `WEB_CRYPTO`, the block above fails if `WEB_CRYPTO` is _disabled_ because the mac crypto libraries are only added when `WEB_CRYPTO` is enabled. So we can either always add the mac crypto libraries when on mac, or fall back to the qt crypto functions when `WEB_CRYPTO` is disabled.

**Comment by annulen** [comment]
Oh, I've missed that. We should use mac crypto libraries when on mac. Could you provide a patch?

**Comment by annulen** [comment]
"Qt crypto" is not really a solution for WebCrypto. It only implements hash sums, which are required for CSP, and cannot do more because Qt doesn't provide any encryption algorithms. However, there is cross-platform GCrypt backend, which implements all necessary algorithms, doesn't have any dependencies and is licensed under LGPL 2.1. This should be a good solution for all other platforms, however on Mac there is native backend so I think we should use it.

**Pull request closed without merging**